lib/icinga/checkable.hpp

```diff
@@ -111,7 +111,14 @@ class Checkable : public ObjectImpl<Checkable>
 
 	void ExecuteRemoteCheck(const Dictionary::Ptr& resolvedMacros = nullptr);
 	void ExecuteCheck();
-	void ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrigin::Ptr& origin = nullptr);
+	enum class ProcessingResult
+	{
+		Ok,
+		NoCheckResult,
+		CheckableInactive,
+		NewerCheckResultPresent,
+	};
+	Checkable::ProcessingResult ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrigin::Ptr& origin = nullptr);
```

Comment:
This is within `Checkable`, so no need to prefix the return type with `Checkable::` here.